hack/lib/test.sh

```diff
@@ -238,6 +238,24 @@ kube::test::if_has_string() {
   fi
 }
 
+kube::test::if_has_not_string() {
+  local message=$1
+  local match=$2
+
+  if [[ $(echo "$message" | grep "$match") ]]; then
```

Comment:
Although I trust your bash knowledge, somehow the original solution seems more readable to me bash scripting ignorant ;)